File: app/src/main/java/com/example/burhan/architectureexample/NoteDao.java
package com.example.burhan.architectureexample;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.*;

@Dao
public interface NoteDao {

    @Insert
    void insert(Note note);

    @Update
    void update(Note note);

    @Delete
    void delete(Note note);

    @Query("DELETE FROM note_table")
    void deleteAllNotes();

    @Query("SELECT * FROM note_table ORDER BY priority DESC")
    LiveData<List<Note>> getAllNotes();

    @Query(" UPDATE note_table SET isComplete = :val WHERE id=:noteID;")
    void setIsComplete(int noteID, boolean val);
}

File: app/src/main/java/com/example/burhan/architectureexample/NoteDao.kt
package com.example.burhan.architectureexample

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface NoteDao {

    @get:Query("SELECT * FROM note_table ORDER BY priority DESC")
    val allNotes: LiveData<List<Note>>

    @Insert
    fun insert(note: Note)

    @Update
    fun update(note: Note)

    @Delete
    fun delete(note: Note)

    @Query("DELETE FROM note_table")
    fun deleteAllNotes()

    @Query(" UPDATE note_table SET isComplete = :checked WHERE id=:noteID;")
    fun setIsComplete(noteID: Int, checked: Boolean)
}